src/devices/Amg88xx/Amg88xx.cs

```diff
@@ -0,0 +1,475 @@
+        }
+
+        /// <summary>
+        /// Gets the interrupt flag from the status register
+        /// </summary>
+        /// <returns>Interrupt flag</returns>
+        public bool HasInterrupt()
+        {
+            return GetBit(Register.STAT, (byte)StatusFlagBit.INTF);
+        }
+
+        /// <summary>
+        /// Clears the interrupt flag in the status register
+        /// </summary>
+        public void ClearInterrupt()
+        {
+            // only the bit to be cleared is set, the other bits need to be 0
+            SetRegister(Register.SCLR, 1 << (byte)StatusClearBit.INTCLR);
+        }
+
+        /// <summary>
+        /// Clears all flags in the status register
+        /// </summary>
+        public void ClearAllFlags()
+        {
+            // only the bit to be cleared is set, the other bits need to be 0
+            SetRegister(Register.SCLR, (1 << (byte)StatusClearBit.OVFCLR) | (1 << (byte)StatusClearBit.OVFTHCLR) | (1 << (byte)StatusClearBit.INTCLR));
+        }
+
+        #endregion
+
+        #region Moving average
+
+        /// <summary>
+        /// Get the state of the moving average mode
+        /// </summary>
+        /// <returns>True, if moving average mode is on</returns>
+        public bool GetMovingAverageModeState()
+        {
+            return GetBit(Register.AVE, (byte)MovingAverageModeBit.MAMOD);
+        }
+
+        /// <summary>
+        /// Sets the moving average mode state.
+        /// </summary>
+        /// <param name="state">True, to switch moving average on</param>
+        public void SetMovingAverageModeState(bool state)
+        {
+            SetBit(Register.AVE, (byte)MovingAverageModeBit.MAMOD, state);
+        }
+
+        #endregion
+
+        #region Frame Rate
+
+        /// <summary>
+        /// Get the current frame rate.
+        /// </summary>
+        /// <returns>Frame rate (either 1 or 10fps) </returns>
+        public int GetFrameRate()
+        {
+            return GetBit(Register.FPSC, (byte)FrameRateBit.FPS) ? 1 : 10;
+        }
+
+        /// <summary>
+        /// Sets the frame rate (either 1 or 10fps).
+        /// </summary>
+        /// <param name="frameRate">Frame rate</param>
+        /// <exception cref="ArgumentException">Thrown when attempting to set a frame rate other than 1 or 10</exception>
+        public void SetFrameRate(int frameRate)
+        {
+            if (frameRate != 1 && frameRate != 10)
+            {
+                throw new ArgumentException("Frame rate must either be 1 or 10.", nameof(frameRate));
+            }
+
+            SetBit(Register.FPSC, (byte)FrameRateBit.FPS, frameRate == 1);
+        }
+        #endregion
+
+        #region Operating Mode / Power Control
+
+        /// <summary>
+        /// Gets the current operating mode
+        /// </summary>
+        /// <returns>Operating mode</returns>
+        public OperatingMode GetOperatingMode()
```

Comment:
Same `OperatingMode` as a property would be perfect as well

Reply:
I changed this into a property.

Reply:
Good. Any new push of your code to see all the changes you've made so far?

Reply:
@Ellerbach I just pushed it, though, I didn't complete all remarks, yet. A few are still to do or to think about. Maybe you can comment on some of the open questions I commented.